Tests/AppTests/SearchTests.swift

```diff
@@ -903,6 +1116,79 @@ class SearchTests: AppTestCase {
+        // Setup
+        let p1 = Package(url: "1", platformCompatibility: [.ios])
+        let p2 = Package(url: "2", platformCompatibility: [.macos])
+        try [p1, p2].save(on: app.db).wait()
+        try Repository(package: p1,
+                       defaultBranch: "main",
+                       name: "1",
+                       owner: "foo",
+                       stars: 10,
+                       summary: "test package").save(on: app.db).wait()
+        try Repository(package: p2,
+                       defaultBranch: "main",
+                       name: "2",
+                       owner: "bar",
+                       summary: "test package").save(on: app.db).wait()
+        try Version(package: p1, packageName: "p1", reference: .branch("main"))
+            .save(on: app.db).wait()
+        try Version(package: p2, packageName: "p2", reference: .branch("main"))
+            .save(on: app.db).wait()
+        try Search.refresh(on: app.db).wait()
+
+        do {
+            // MUT
+            let res = try Search.fetch(app.db, ["test", "stars:>5"], page: 1, pageSize: 20).wait()
+
+            // validate
+            XCTAssertEqual(res.results.compactMap(\.packageResult?.repositoryName), ["1"])
+        }
+
+        do {  // double check that leaving the filter term off selects both packages
+            // MUT
+            let res = try Search.fetch(app.db, ["test"], page: 1, pageSize: 20).wait()
+
+            // validate
+            XCTAssertEqual(
+                res.results.compactMap(\.packageResult?.repositoryName).sorted(),
+                ["1", "2"]
+            )
+        }
+    }
+
+    func test_SearchFilter_error() throws {
+        // Test error handling in case of an invalid filter
+        // Setup
+        let p1 = Package(url: "1", platformCompatibility: [.ios])
+        let p2 = Package(url: "2", platformCompatibility: [.macos])
+        try [p1, p2].save(on: app.db).wait()
+        try Repository(package: p1,
+                       defaultBranch: "main",
+                       license: .mit,
+                       name: "1",
+                       owner: "foo",
+                       summary: "test package").save(on: app.db).wait()
+        try Repository(package: p2,
+                       defaultBranch: "main",
+                       license: .none,
+                       name: "2",
+                       owner: "bar",
+                       summary: "test package").save(on: app.db).wait()
+        try Version(package: p1, packageName: "p1", reference: .branch("main"))
+            .save(on: app.db).wait()
+        try Version(package: p2, packageName: "p2", reference: .branch("main"))
+            .save(on: app.db).wait()
+        try Search.refresh(on: app.db).wait()
+
+        // MUT
+        let res = try Search.fetch(app.db, ["test", "license:>mit"], page: 1, pageSize: 20).wait()
+
+        // validate
+        XCTAssertEqual(res.results.compactMap(\.packageResult?.repositoryName), [])
```

Comment:
We may want to adjust this behaviour: right now it simply swallows the error and returns an empty list without any indication that the filter is invalid:

<img width="491" alt="CleanShot 2022-01-06 at 08 59 17@2x" src="https://user-images.githubusercontent.com/65520/148349682-241ecdb7-3962-4250-98cb-fb4d497165d7.png">

Reply:
Oh yea. I think this should be logged as a bug. Maybe we need a set of operators that are valid for each type of filter field, or something like that.